File: sdk/csharp/src/Providers/TempMailPlus.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;
using System.Text.Json.Nodes;

namespace XxxXTeam.TempMail.Providers;

/// <summary>
/// TempMail Plus 渠道家族 — https://tempmail.plus
/// 无需认证、无需 Cookie、无需 Token 的简单 REST API；默认域名 mailto.plus，
/// fexpost.com / fexbox.org / mailbox.in.ua 等别名仅换域名与渠道标识。
/// </summary>
public static class TempMailPlus
{
    private const string ApiBase = "https://tempmail.plus/api/mails";
    private const string DefaultDomain = "mailto.plus";
    private const string DefaultChannel = "tempmail-plus";
    private static readonly Random Rand = new();

    private static readonly Dictionary<string, string> Headers = new()
    {
        ["Accept"] = "application/json",
        ["User-Agent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36",
        ["Referer"] = "https://tempmail.plus/",
        ["Origin"] = "https://tempmail.plus",
    };

    private static string RandomLocal(int length = 12)
    {
        const string chars = "abcdefghijklmnopqrstuvwxyz0123456789";
        var sb = new StringBuilder();
        for (var i = 0; i < length; i++) sb.Append(chars[Rand.Next(chars.Length)]);
        return sb.ToString();
    }

    /// <summary>创建 tempmail-plus 临时邮箱（指定域名与渠道标识）</summary>
    public static EmailInfo Generate(string? domain = null, string? channel = null)
    {
        var selectedDomain = string.IsNullOrWhiteSpace(domain) ? DefaultDomain : domain.Trim();
        var selectedChannel = string.IsNullOrWhiteSpace(channel) ? DefaultChannel : channel.Trim();
        var email = $"{RandomLocal()}@{selectedDomain}";

        // 调用列表接口验证地址可用
        var resp = Http.Get($"{ApiBase}/?email={WebUtility.UrlEncode(email)}&epin=", Headers);
        resp.EnsureSuccess();

        return new EmailInfo(selectedChannel, email);
    }

    private static JsonObject? FetchBody(string mailId, string email)
    {
        if (string.IsNullOrEmpty(mailId)) return null;
        try
        {
            var resp = Http.Get($"{ApiBase}/{mailId}?email={WebUtility.UrlEncode(email)}&epin=", Headers);
            if (!resp.Ok) return null;
            return Json.Parse(resp.Body) as JsonObject;
        }
        catch { return null; }
    }

    /// <summary>获取 tempmail-plus 邮件列表（逐封拉取正文）</summary>
    public static List<Email> GetEmails(string email)
    {
        var e = (email ?? "").Trim();
        if (e.Length == 0) throw new Exception("tempmail-plus: 邮箱地址为空");

        var resp = Http.Get($"{ApiBase}/?email={WebUtility.UrlEncode(e)}&epin=", Headers);
        resp.EnsureSuccess();
        var rows = (Json.Parse(resp.Body) as JsonObject)?["mail_list"] as JsonArray;
        var result = new List<Email>();
        if (rows is null) return result;

        foreach (var raw in rows)
        {
            if (raw is not JsonObject ro) continue;
            var mailId = Json.Str(ro, "mail_id");
            var detail = FetchBody(mailId, e);

            var from = Json.Str(ro, "from_mail");
            if (string.IsNullOrEmpty(from)) from = Json.Str(ro, "from_name");

            result.Add(Normalize.NormalizeEmail(new Dictionary<string, object?>
            {
                ["id"] = mailId,
                ["from"] = from,
                ["to"] = e,
                ["subject"] = Json.Str(ro, "subject"),
                ["date"] = Json.Str(ro, "time"),
                ["html"] = detail is null ? "" : Json.Str(detail, "html"),
                ["text"] = detail is null ? "" : Json.Str(detail, "text"),
                ["isRead"] = !((ro["is_new"] as JsonValue)?.GetValue<bool>() ?? true),
            }, e));
        }
        return result;
    }
}

File: sdk/php/src/Providers/TempMailPlus.php
<?php

declare(strict_types=1);

namespace ChanhanzhanX\TempMail\Providers;

use ChanhanzhanX\TempMail\Email;
use ChanhanzhanX\TempMail\EmailInfo;
use ChanhanzhanX\TempMail\HttpClient;
use ChanhanzhanX\TempMail\Normalize;

/**
 * TempMail Plus 渠道实现
 *
 * https://tempmail.plus，无需认证的简单 REST API，默认域名 mailto.plus。
 */
final class TempMailPlus
{
    private const API_BASE = 'https://tempmail.plus/api/mails';
    private const DOMAIN = 'mailto.plus';

    /** @var array<string,string> */
    private const HEADERS = [
        'Accept' => 'application/json',
        'User-Agent' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36',
        'Referer' => 'https://tempmail.plus/',
        'Origin' => 'https://tempmail.plus',
    ];

    private static function randomLocal(int $length = 12): string
    {
        $chars = 'abcdefghijklmnopqrstuvwxyz0123456789';
        $out = '';
        for ($i = 0; $i < $length; $i++) {
            $out .= $chars[random_int(0, strlen($chars) - 1)];
        }
        return $out;
    }

    public static function generate(?string $domain = null, string $channel = 'tempmail-plus'): EmailInfo
    {
        $selectedDomain = trim((string) ($domain ?? self::DOMAIN)) ?: self::DOMAIN;
        $selectedChannel = trim($channel) ?: 'tempmail-plus';
        $email = self::randomLocal() . '@' . $selectedDomain;
        // 调用列表接口验证地址可用
        HttpClient::get(self::API_BASE . '/?email=' . rawurlencode($email) . '&epin=', self::HEADERS);
        return new EmailInfo($selectedChannel, $email, $email);
    }

    /**
     * @return Email[]
     */
    public static function getEmails(string $email, ?string $token): array
    {
        $e = trim($email);
        if ($e === '') {
            throw new \InvalidArgumentException('tempmail-plus: 邮箱地址为空');
        }
        $resp = HttpClient::get(self::API_BASE . '/?email=' . rawurlencode($e) . '&epin=', self::HEADERS);
        $data = HttpClient::json($resp);
        $rows = $data['mail_list'] ?? null;
        if (!is_array($rows)) {
            return [];
        }
        $out = [];
        foreach ($rows as $raw) {
            if (!is_array($raw)) {
                continue;
            }
            $mailId = $raw['mail_id'] ?? 0;
            $detail = self::fetchBody($mailId, $e);
            $out[] = Normalize::email([
                'id' => $mailId ? (string) $mailId : '',
                'from' => $raw['from_mail'] ?? $raw['from_name'] ?? '',
                'to' => $e,
                'subject' => $raw['subject'] ?? '',
                'date' => $raw['time'] ?? '',
                'html' => $detail['html'] ?? '',
                'text' => $detail['text'] ?? '',
                'isRead' => !($raw['is_new'] ?? true),
            ], $e);
        }
        return $out;
    }

    /**
     * @return array<mixed>
     */
    private static function fetchBody(mixed $mailId, string $email): array
    {
        if (!$mailId) {
            return [];
        }
        try {
            $resp = HttpClient::get(self::API_BASE . '/' . rawurlencode((string) $mailId) . '?email=' . rawurlencode($email) . '&epin=', self::HEADERS);
            $data = HttpClient::json($resp);
            return is_array($data) ? $data : [];
        } catch (\Throwable) {
            return [];
        }
    }
}
